### PR TITLE
feat: 🎸 target list will update according to org scope selected

### PR DESCRIPTION
**Pull request by cameronperera**
✅ Closes: https://hashicorp.atlassian.net/browse/ICU-12205

:tickets: [Jira ticket](https://hashicorp.atlassian.net/browse/ICU-12205)

## Description

If a user selects an org it will now only show targets that belong to that org scope.

## Screenshots (if appropriate):

https://github.com/hashicorp/boundary-ui/assets/29386339/e09445a0-f280-41af-a5a8-4b6240502ebf

## How to Test

Run `ENABLE_MIRAGE=false BYPASS_CLI_SETUP=true yarn start:desktop` to run against a local binary built off the main branch of the boundary repo. Create a second org with a project and target for listing purposes. After loading all targets when logging in from the global scope, select the second org in the header of the desktop client. The targets listed should only belong to the org you selected from the header of the desktop client.

## Checklist:

- [x] I have added before and after screenshots for UI changes
- [ ] ~I have added JSON response output for API changes~
- [x] I have added steps to reproduce and test for bug fixes in the description
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js [hunk 1]
```diff
@@ -84,11 +84,18 @@ export default class ScopesScopeProjectsTargetsIndexRoute extends Route {
     transition,
```
Comment (lisbet-alvarez) [inline ui/desktop/app/routes/scopes/scope/projects/targets/index.js:84]: non-blocking question: should the sessions route be updated  as well or will that be in a separate PR ?

**Review by lisbet-alvarez — APPROVED** [review]
LGTM

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js [hunk 2]
```diff
@@ -84,11 +84,18 @@ export default class ScopesScopeProjectsTargetsIndexRoute extends Route {
...
   ) {
     await this.getAllTargets(transition);
+    const projects = this.modelFor('scopes.scope.projects');
+    const projectIds = projects.map((project) => project.id);
 
     const filters = { scope_id: [], id: { values: [] }, type: [] };
     scopes.forEach((scope) => {
       filters.scope_id.push({ equals: scope });
     });
+    if (scopes.length === 0) {
```
Comment (ZedLi) [inline ui/desktop/app/routes/scopes/scope/projects/targets/index.js:94]: One issue is this assumes that the client daemon is up...which I think is fine for now

**Review by ZedLi — APPROVED** [review]

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js (hunk 2 above)
Comment (cameronperera) [reply ui/desktop/app/routes/scopes/scope/projects/targets/index.js:94]: In one of your PRs you add the `isClientDaemonRunning`. I could utilize that here to bypass this step.

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js (hunk 1 above)
Comment (cameronperera) [reply ui/desktop/app/routes/scopes/scope/projects/targets/index.js:84]: That is a good point. I will add it to sessions now!

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js (hunk 2 above)
Comment (ZedLi) [reply ui/desktop/app/routes/scopes/scope/projects/targets/index.js:94]: Sure but you'd need to manually filter it down then

File: ui/desktop/app/routes/scopes/scope/projects/targets/index.js (hunk 1 above)
Comment (cameronperera) [reply ui/desktop/app/routes/scopes/scope/projects/targets/index.js:84]: Interesting. I added this same logic to the sessions route. However, when you change the scope at the top of the app, it redirects you to the targets route.

Comment (cameronperera) [reply ui/desktop/app/routes/scopes/scope/projects/targets/index.js:84]: Updated!

**Review by ZedLi — APPROVED** [review]
Test sessions as well and works great!
